stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+                        self.sub_resource += Textures.NORMAL_OPTIONS + f"normal_texture = ExtResource( {self.ext_resource_id} )\n"
+                    elif jpg == "depth":
+                        self.sub_resource += Textures.DEPTH_OPTIONS + f"depth_texture = ExtResource( {self.ext_resource_id} )\n"
+                    # add .jpg as ext_resource
+                    self.ext_resource += f'[ext_resource path="res://{Textures.TEXTURE_FOLDER}/{t_info.folder}/{value}" type="Texture" id={self.ext_resource_id}]\n'
+                    self.ext_resource_id += 1
+
+                # add uv1_scale if texture has it
+                if t_info.uv1_scale:
+                    self.sub_resource += f"uv1_scale = {t_info.uv1_scale}\n"
+
+            # handle textures which don't need .jpgs (only need sub_resource)
+            elif texture in Textures.other_textures:
+                self.sub_resource += f'[sub_resource type="SpatialMaterial" id={self.sub_resource_id}]\n{Textures.other_textures[texture]}'
+
+            self.sub_resource += "\n"
+                
+            # add the sub resource's index to texture_index so we can use it later 
+            self.texture_index[texture] = self.sub_resource_id
+            self.sub_resource_id += 1 
+
+    def add_obj_file(self, stl_file_extless, uv_map, mesh_compression, texture):
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .stl into bpy
+        bpy.ops.import_mesh.stl(filepath = f"{self.input_folder}/{stl_file_extless}.stl")
+        # Get all objects in selection
+        selection = bpy.context.selected_objects
+        # Deselect all objects
+        bpy.ops.object.select_all(action='DESELECT')
+
+        for obj in selection:
+            # Select each object
+            obj.select_set(True)
+            # Make it active
+            bpy.context.view_layer.objects.active = obj
+            # Toggle into Edit Mode
+            bpy.ops.object.mode_set(mode='EDIT')
+            # Select the geometry
+            bpy.ops.mesh.select_all(action='SELECT')
+
+            # Remove unnecessary faces 
+            if mesh_compression == "limited_dissolve": 
+                bpy.ops.mesh.dissolve_limited()
+            elif mesh_compression == "tris_to_quads":
+                bpy.ops.mesh.tris_convert_to_quads() 
+
+            # Call project operator to generate uv maps
+            try:
+                UV_MAPS[uv_map]()
+            except KeyError as e:
+                print(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
+
+            # Toggle out of Edit Mode
+            bpy.ops.object.mode_set(mode='OBJECT')
+            # Deselect the object
+            obj.select_set(False)
+
+        # Export to obj
+        bpy.ops.wm.obj_export(filepath = f"{self.output_folder}/{stl_file_extless}.obj")
+        # remove the .mtl file
+        os.remove(f"{self.output_folder}/{stl_file_extless}.mtl")
```

Comment:
More opportunities to use `pathlib` to allow platform-independent use. There are other places to use it in this code, but I'll avoid pointing them all out for brevity.
```suggestion
        export_path = Path(self.output_folder) / Path(stl_file_extless).with_suffix('.obj')
        bpy.ops.wm.obj_export(filepath = str(export_path))
        
        # remove the .mtl file
        mtl_filepath = Path(self.output_folder) / Path(stl_file_extless).with_suffix('.mtl')
        Path.unlink(mtl_filepath)
```